tests/test_coverage.py

```diff
@@ -0,0 +1,181 @@
+    with tempfile.NamedTemporaryFile(suffix=".db") as tmp_db:
+        method(
+            database=tmp_db.name,
+            fasta=fasta_dir,
+            name="Artificial",
+            create_db=True,
+            **kwargs,
+        )
+        output = capsys.readouterr().out
+        assert " run-id 1 with " in output
+
+        session = db_orm.connect_to_db(tmp_db.name)
+        run = session.query(db_orm.Run).one()
+        session.close()
+        return run
+
+
+def test_coverage(capsys: pytest.CaptureFixture[str], tmp_path: str) -> None:
+    """Check comparison have expected identity and query-coverage."""
+    tmp_dir = Path(tmp_path)
+    seq_dir = tmp_dir / "fasta"
+    seq_dir.mkdir()
+    (seq_dir / "small.fasta").symlink_to(
+        Path("tests/fixtures/MIBY01000005.fasta").resolve()
+    )  # 154173fb8e7415ab45532a738572f957 - 7582 bp
+    (seq_dir / "large.fasta").symlink_to(
+        Path("tests/fixtures/MIBY01000011.fasta").resolve()
+    )  # a0efc718e680e34d2f5c8f5d2286ca9c - 18001 bp
+    with (seq_dir / "both.fasta").open("w") as handle:
+        for file in (
+            "tests/fixtures/MIBY01000005.fasta",
+            "tests/fixtures/MIBY01000011.fasta",
+        ):
+            with Path(file).open() as in_handle:
+                for line in in_handle:
+                    handle.write(line)
+        # Gives 7b6a6226ce00e52edca15565aa0d270d for both contigs
+    assert len(list(seq_dir.glob("*.f*"))) == 3  # noqa: PLR2004
+
+    checksums = (
+        '["154173fb8e7415ab45532a738572f957",'  # small contig
+        '"7b6a6226ce00e52edca15565aa0d270d",'  # both contigs
+        '"a0efc718e680e34d2f5c8f5d2286ca9c"]'  # large contig
+    )
+    # Small vs large (and large vs small) should fail, giving null.
+    # Self-vs-self ought to give 100% but we picked these two genomes
+    # because they were odd for self-vs-self, so expect 99~100% identity:
+    #
+    # i.e. [[99~100%, 99~100%,    null],
+    #       [99~100%, 99~100%, 99~100%],
+    #       [   null, 99~100%, 99~100%]] for identity
+    #
+    # In terms of coverage, this should be 100% on the diagonal.
+    # For the first row (small query) either 100% (vs self of the both file)
+    # or null (vs the larger contig). For the middle row (both contigs),
+    # coverage will be ~30% (vs small), 100% (vs self), and ~70% (vs large).
+    # Finally, for the last row (large query), coverage will be null (vs
+    # small), 100% (vs both), and 100% (vs self).
+    #
+    # i.e. [[100%, 100%, null],
+    #       [~30%, 100%, ~70%],
+    #       [null, 100%, 100%]] for query_cov
+    #
+    # Using the contig lengths, small is 7582 bp, large is 18001 bp, gives
+    # the expected coverages values vs the combined file of 29.6% and 70.4%.
+    #
+    # Comparing the JSON serialised dataframe directly from the DB
+
+    run = do_comparison(capsys, seq_dir, public_cli.anim)
+    assert run.df_identity == (
+        "{"
```

Comment:
The ``{`` is a string on its own because it messes up as part of an f-string. There could be a better solution than this?